renderdoc/os/win32/win32_shellext.cpp

```diff
@@ -362,18 +399,219 @@ struct RDCThumbnailProvider : public IThumbnailProvider, IInitializeWithStream
       return E_NOTIMPL;
     }
 
-    const byte *jpgbuf = m_Thumb.pixels;
     size_t thumblen = m_Thumb.len;
     uint32_t thumbwidth = m_Thumb.width, thumbheight = m_Thumb.height;
+    byte *thumbpixels = nullptr;
 
-    if(jpgbuf == NULL)
-      return E_NOTIMPL;
+    if(m_Thumb.format == FileType::JPG)
+    {
+      const byte *jpgbuf = m_Thumb.pixels;
+      if(jpgbuf == NULL)
+        return E_NOTIMPL;
+
+      int w = thumbwidth;
+      int h = thumbheight;
+      int comp = 3;
+      thumbpixels = jpgd::decompress_jpeg_image_from_memory(jpgbuf, (int)thumblen, &w, &h, &comp, 3);
+    }
+    else
+    {
+      const ResourceFormatType compType = m_ddsData.format.type;
+      const uint32_t decompressedStride = thumbwidth * 4;
+      const uint32_t decompressedBlockWidth = 16;    // in bytes (4 byte/pixel)
+      const uint32_t decompressedBlockMaxSize = 64;
+
+      ResourceFormatType ddsFormatType = ResourceFormatType::BC1;
+      int compressedBlockSize = 8;
+      switch(compType)
```

Comment:
The assignment here of `ddsFormatType` in each case seems to be redundant unless I'm missing something, since it is always equal to `compType`. There's also a helper, `ResourceFormat::ElementSize()`, which returns the byte size of a block for block-compressed formats. I think then the switch can be simplified to only check if the type is one we support (BC1-BC5) and return `E_NOTIMPL` otherwise.

Reply:
Thank you, that is a useful helper. Changed to `compressedBlockSize = ResourceFormat::ElementSize()` to get the byte size of a block, which made the switch statement redundant, so that is removed now.
'ddsFormatType' had no usage so that is removed too.